int8   x
int8   y
int32  theta